File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cs
using System;
using System.Text;

public class Solution {
    public string solution(string number, int k) {
        var sb = new StringBuilder(number.Length);
        
        foreach(char c in number)
        {
            while(k > 0 && sb.Length > 0 && sb[sb.Length - 1] < c)
            {
                sb.Length--;
                k--;
            }
            sb.Append(c);
        }
        
        if(k > 0)
        {
            sb.Length -= k;
        }
        return sb.ToString();
    }
}

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(string number, int k) {
    stack<char> stk;
    
    for(char c : number)
    {
        while(stk.size() && k > 0 && stk.top() < c)
        {
            stk.pop();
            k--;
        }
        stk.push(c);
    }
    
    while(k > 0)
    {
        stk.pop();
        k--;
    }
    
    string answer = "";
    while(stk.size())
    {
        answer += stk.top();
        stk.pop();
    }
    reverse(answer.begin(), answer.end());
    return answer;
}